src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -84,38 +84,58 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
 }, {});
 
 const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelect }) => {
-  const [isNewSelectedPath, setIsNewSelectedPath] = useState(false);
-  const [previousSelectedPath, setPreviousSelectedPath] = useState(selectedPath);
   const [openKeys, setOpenKeys] = useState(openKeysToItem(menuItems[0], selectedPath));
+  const [currentNodeId, setCurrentNodeId] = useState();
+  const cursor = useRef(0);
   const selectedItem = useRef();
   const theme = useContext(ThemeContext);
+  const visibleNodes = [];
 
-  if (previousSelectedPath !== selectedPath) {
-    setIsNewSelectedPath(true);
-    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
-    setPreviousSelectedPath(selectedPath);
-  }
-
+  /**
+   * Scrolls the currently selected menu item into view on mount.
+   */
   useEffect(() => {
     if (selectedItem && selectedItem.current) {
       selectedItem.current.scrollIntoView();
     }
   }, []);
 
-  useLayoutEffect(() => {
-    if (!isNewSelectedPath) {
-      return;
-    }
-    const currentItemPosition = selectedItem?.current ? selectedItem.current.getBoundingClientRect() : null;
+  /**
+   * Updates cursor position when new menu items are loaded.
+   * If the item emphasized through keyboard navigation is not visible, scroll it into view.
+   */
+  useEffect(() => {
+    cursor.current = visibleNodes.findIndex((el) => el.id === currentNodeId);
+    const currentNode = currentNodeId ? document.querySelector(`#${currentNodeId}`) : null;
+    const currentNodePosition = currentNode ? currentNode.getBoundingClientRect() : null;
     const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
-    if (!currentItemPosition || !navigationMenuPosition) {
-      return;
+
+    if (currentNode) {
+      currentNode.focus();
+    }
+    if (currentNode && currentNodePosition && (currentNodePosition.bottom > navigationMenuPosition.bottom || currentNodePosition.top < navigationMenuPosition.top)) {
+      currentNode.scrollIntoView();
     }
-    // If the current item is not visible, scroll the item into view.
-    if (currentItemPosition.bottom > navigationMenuPosition.bottom || currentItemPosition.top < navigationMenuPosition.top) {
+  }, [currentNodeId, visibleNodes]);
```

Comment:
`visibleNodes` is instantiated every render, using it as a key will cause this effect to be invoked on every update

Reply:
Removing it from the dependency list requires me to disable a linting rule for this line. I'm assuming that's ok in this case?